### DISCUSSION TITLE
problem in DMA test

### DISCUSSION THREAD

**Discussion by 250Gamer** [Bodmer/TFT_eSPI#1869] [Q&A - General] [answered]
I tried all the examples in DMA test
but there were always a problem
here are the error information：
g:/arduino/hardware/espressif/esp32/tools/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: sketch\boing_ball.ino.cpp.o:(.literal._Z5setupv+0x24): undefined reference to `TFT_eSPI::initDMA(bool)'
g:/arduino/hardware/espressif/esp32/tools/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: sketch\boing_ball.ino.cpp.o:(.literal._Z4loopv+0x8c): undefined reference to `TFT_eSPI::pushPixelsDMA(unsigned short*, unsigned int)'
g:/arduino/hardware/espressif/esp32/tools/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: sketch\boing_ball.ino.cpp.o: in function `setup()':
G:\Arduino\libraries\TFT_eSPI-master\examples\DMA test\boing_ball/boing_ball.ino:66: undefined reference to `TFT_eSPI::initDMA(bool)'
g:/arduino/hardware/espressif/esp32/tools/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: sketch\boing_ball.ino.cpp.o: in function `loop()':
G:\Arduino\libraries\TFT_eSPI-master\examples\DMA test\boing_ball/boing_ball.ino:163: undefined reference to `TFT_eSPI::pushPixelsDMA(unsigned short*, unsigned int)'
collect2.exe: error: ld returned 1 exit status
exit status 1
I need help
![Snipaste_2022-06-06_09-46-15](https://user-images.githubusercontent.com/106938524/172081243-aa563408-0047-4180-be32-06020e77ac6f.jpg)
![Snipaste_2022-06-06_09-46-15](https://user-images.githubusercontent.com/106938524/172081321-d997f856-c206-45f6-a7d8-d73e08d5497c.jpg)

**Comment by Bodmer** [comment, marked as answer]
I ran a test with only setup 23 selected in User_Setup_Select.h and compiled with the ESP32 Dev Module selected and it compiled fine.

DMA is not supported for later variants of the ESP32, e.g. ESP32 C3/S2/S3. Furthermore the ESP32 C2 is not supported.

The original ESP32 boards work fine.

If you are stil having trouble then I need a lot more detail on your hardware and software setup, including software versions.
